app/Services/CreateConferenceForm.php

```diff
@@ -0,0 +1,50 @@
+<?php namespace SaveMyProposals\Services;
+
+use Conference;
+use SaveMyProposals\Exceptions\ValidationException;
+use Validator;
+
+class CreateConferenceForm
+{
+    private $rules = [
+        'title' => ['required'],
+        'description' => ['required'],
+        'url' => ['required'],
+        'starts_at' => ['date'],
+        'ends_at' => ['date'],
+        'cfp_starts_at' => ['date'],
+        'cfp_ends_at' => ['date'],
+    ];
+
+    private $input;
+    private $user;
+
+    private function __construct($input, $user)
+    {
+        $this->input = $this->removeEmptyFields($input);
+        $this->user = $user;
+    }
+
+    private function removeEmptyFields($input)
```

Comment:
Yeah, that will strip out anything that comes in as an empty string. As I'm pretty sure you noticed, submitting a `starts_at` as empty for example will actually trigger MySQL into storing `0000-00-00` even though the default value for the column is `NULL`.

This is all only for brand new conferences though, not updating existing ones, all that is still in the controller in the other action.